src/hooks/useRequestParams.ts

```diff
@@ -1,93 +1,108 @@
+  chainId: ss.string(),
+  oracleType: ss.optional(ss.string()),
+});
 
-export type ActiveRequestParams = {
+type RequestInputRequired = {
+  chainId: number;
+  oracleType: OracleType;
+};
+type RequestInput = RequestInputRequired & {
   requester: string;
+  timestamp: number;
   identifier: string;
   ancillaryData: string;
-  chainId: number;
-  timestamp: number;
 };
-export type ActiveRequestParamsByTransaction = {
-  chainId: number;
+type RequestInputByTransaction = RequestInputRequired & {
+  eventIndex: number;
   transactionHash: string;
-  eventIndex: number | undefined;
 };
 
-type RequestParams<IsByTransaction extends boolean> =
-  IsByTransaction extends false
-    ? ActiveRequestParams
-    : ActiveRequestParamsByTransaction;
-type Keys<IsByTransaction extends boolean> = IsByTransaction extends false
-  ? typeof SEARCH_PARAMS_KEYS
-  : typeof SEARCH_PARAMS_KEYS_BY_TRANSACTION;
-type UseRequestParamsReturn<IsByTransaction extends boolean> =
-  | { request: RequestParams<IsByTransaction>; error: undefined }
-  | { request: undefined; error: InvalidURLParamsError };
-export function useRequestParams(): UseRequestParamsReturn<true>;
-export function useRequestParams<T extends boolean>(
-  isByTransaction: T
-): UseRequestParamsReturn<T>;
-export default function useRequestParams<T extends boolean>(
-  isByTransaction?: T
-): UseRequestParamsReturn<T> {
-  const [searchParams] = useSearchParams();
-  const keys = (
-    Boolean(isByTransaction)
-      ? SEARCH_PARAMS_KEYS_BY_TRANSACTION
-      : SEARCH_PARAMS_KEYS
-  ) as Keys<T>;
-
-  const request = useMemo(() => {
-    const params = {} as any;
-    try {
-      keys.forEach((key) => {
-        const value = searchParams.get(key);
-        if (value === null && key !== "eventIndex") {
-          throw new InvalidURLParamsError(`Missing required parameter: ${key}`);
-        }
-        if (["timestamp", "chainId", "eventIndex"].includes(key)) {
-          const parsedKey = Number(value);
-          params[key] = isNaN(parsedKey) ? undefined : parsedKey;
-        } else {
-          params[key] = value;
-        }
-      });
-      return { request: params as RequestParams<T>, error: undefined };
-    } catch (e) {
-      return { request: undefined, error: e as InvalidURLParamsError };
-    }
-  }, [keys, searchParams]);
-  return request;
+export function getRequestInputByTransaction(
+  params: unknown
+): RequestInputByTransaction | undefined {
+  try {
+    const parsed = ss.create(params, RequestInputByTransactionSs);
+    parsed.oracleType = parsed.oracleType || defaultOracleType;
+    assert(
+      isSupportedOracleType(parsed.oracleType),
+      "Invalid oracle type in search params: " + parsed.oracleType
+    );
+    const oracleType: OracleType = parsed.oracleType;
+    return {
+      ...parsed,
+      oracleType,
+      chainId: Number(parsed.chainId),
+      eventIndex: Number(parsed.eventIndex),
+    };
+  } catch (err) {
+    // do nothing
```

Comment:
for now we cant really handle errors in hooks, so if this is unable to parse for any reason, just return undefined